### DISCUSSION TITLE
What is the best way to append content to each page without adding a macro to each individual page?

### DISCUSSION THREAD

**Discussion by millerbyte** [fralau/mkdocs-macros-plugin#102] [Q&A] [answered]
I'm looking into using the `on_post_page_macros` hook, but I am unable to append content to the `env.raw_markdown` string:

```
AttributeError: can’t set attribute
```

I realize that this is probably because `raw_markdown` has no setter, but the documentation doesn't go into detail on how to actually manipulate `raw_markdown`.  

I'd like to add a footer to each page with the `git.date` data.  I'd rather not have to add an `{{ add_footer() }}` macro to each page.  Any ideas?

**Comment by Jc-L** [comment]
If it is to apply on all pages, couldn't you add the `{{ add_footer() }}` in a [mkdoc template](https://www.mkdocs.org/dev-guide/themes/#template-files) ?

**Comment by fralau** [comment, marked as answer]
This issue was fixed in #98.
